### PR TITLE
docs: fix description of cluster dns resolution counters #11305

### PR DESCRIPTION
**Pull request by erikbos**
Commit Message: fix description of cluster dns resolution counters
Risk Level: Low
Docs Changes: fixes the description of cluster dns resolution counters based upon input from #11305
Fixes: #11305

### CODE REVIEW

File: docs/root/configuration/upstream/cluster_manager/cluster_stats.rst [hunk 1]
```diff
@@ -89,9 +89,9 @@ Every cluster has a statistics tree rooted at *cluster.<name>.* with the followi
   membership_total, Gauge, Current cluster membership total
   retry_or_shadow_abandoned, Counter, Total number of times shadowing or retry buffering was canceled due to buffer limits
   config_reload, Counter, Total API fetches that resulted in a config reload due to a different config
-  update_attempt, Counter, Total cluster membership update attempts
-  update_success, Counter, Total cluster membership update successes
-  update_failure, Counter, Total cluster membership update failures
+  update_attempt, Counter, Total DNS resolution attempts
```
Comment (ramaraochavali) [inline docs/root/configuration/upstream/cluster_manager/cluster_stats.rst:92]: DNS resolution attempts are  only applicable for STRICT_DNS, LOGICAL_DNS type of clusters. You will have to qualify that

Comment (mattklein123) [reply docs/root/configuration/upstream/cluster_manager/cluster_stats.rst:92]: I think this is also used by redis cluster, so it's a little confusing. Per @ramaraochavali maybe generalize somehow?

Comment (erikbos) [reply docs/root/configuration/upstream/cluster_manager/cluster_stats.rst:92]: Thanks for pointing that out.. are these counters shared with anything else than redis cluster? 

What about:
```suggestion
  update_attempt, Counter, Total cluster membership update attempts. Origin can be DNS resolution or redis cluster
```

(Not sure if we should include the exact configuration scope for when each metric applies?)

Comment (mattklein123) [reply docs/root/configuration/upstream/cluster_manager/cluster_stats.rst]: I would prefer that we don't actually reference specific implementations in the generic cluster docs. Perhaps we can say something like this counters are service discovery method specific and then maybe have some documentation in the docs specific to the types?
Either here: https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/service_discovery
Or maybe broken out in some way within cluster_stats.rst? 

wdyt?

/wait

Comment (erikbos) [reply docs/root/configuration/upstream/cluster_manager/cluster_stats.rst]: I looked at other pages describing statistics, and they tend to have no additional paragraphs, just a description per field. 

The service_discovery page looks like a good candidate, I was not aware of that one. That is a better place to mention these counters as it also describe the DNS behaviour.

Comment (erikbos) [reply docs/root/configuration/upstream/cluster_manager/cluster_stats.rst]: Perhaps also add a link on docs/envoy/latest/configuration/upstream/upstream to docs/envoy/latest/intro/arch_overview/upstream/upstream?

Comment (erikbos) [reply docs/root/configuration/upstream/cluster_manager/cluster_stats.rst]: > Perhaps also add a link on docs/envoy/latest/configuration/upstream/upstream to docs/envoy/latest/intro/arch_overview/upstream/upstream?

Perhaps skip this.. the other configuration section do not appear to link to corresponding architecture sections either.

Comment (erikbos) [reply docs/root/configuration/upstream/cluster_manager/cluster_stats.rst]: What about just:

> update_attempt, Counter, Total cluster membership update attempts by service discovery.

**Comment by mattklein123** [comment]
Thanks LGTM, can you fix DCO? You will need to squash and force push.

/wait

**Pull request merged**